Deck.cpp

```diff
@@ -0,0 +1,72 @@
+#include "Deck.h"
+
+void Deck::createDeckandShuffle()
+{
+  for(int i{0}; i < m_deck.size(); i++)
+  {
+    m_deck.at(i).rank = static_cast<Card::Rank>(i%13);
+    m_deck.at(i).suit = static_cast<Card::Suit>(i/13);
+  }
+  static std::mt19937 mt{static_cast<std::mt19937::result_type>(std::time(nullptr))};
+  std::shuffle(m_deck.begin(), m_deck.end(), mt);
+}
+
+int Deck::getCardValue()
+{
+    int cardValue{};
+    Card::Rank rankForComparison{};
+    if(std::find(m_deck.begin(), m_deck.end(), m_deck[m_index]) == std::end(m_deck))
+      assert(false && "Invalid index");
+    else
+      rankForComparison = m_deck[m_index].rank;
+    switch(rankForComparison)
+    {
+        case (Card::Rank::two) : cardValue = 2; break;
```

Comment:
Huh, I actually didn't know you could assign values to enums like that. Seems sort of obvious now looking back.

> Since both schemes make equal sense I'd say don't do either and leave it how you have it, sort of as a middle ground if that makes sense

Sounds good